index.js

```diff
@@ -1 +1,97 @@
 // implement your API here
+const express = require('express') //import express
+
+const Users = require("./data/db.js") //import server
+
+const server = express(); //start server
+
+server.use(express.json()); //tells server to parse JSON
+
+//GET function returns all users.
+server.get("/users", (req, res) => {
+    Users.find()
+        .then(user => {
+            res.status(200).json(user);
+        })
+        .catch(err => {
+            res.status(500).json({ message: "The users information could not be retrieved." })
+        })
+})
+
+//GET function returns specific ID
+server.get("/users/:id", (req, res) => {
+    const { id } = req.params;
+
+    Users.findById(id)
+        .then(user => {
+            if (id) {
+                res.status(200).json(user)
+            } else {
+                res.status(404).json({ message: "User not found" })
+            }
+        })
+        .catch(err => {
+            res.status(500).json({ error: "The user information could not be retrieved." })
+        })
+})
+
+//POST function to add a new user 
+server.post('/users', (req, res) => {
+    const userInfo = req.body;
+    console.log(userInfo);
+
+    Users.insert(userInfo)
+        .then(person => {
+            if (userInfo.name && userInfo.bio) {
+                Users.findById(person.id)
+                    .then(user => {
+                        res.status(200).json(user)
+                    })
+            } else {
+                res.status(400).json({ message: "Name and Bio are required" })
+            }
+        })
+        .catch(err => {
+            res.status(500).json(err);
+        });
+});
+
+//DELETE function to delete a user
+server.delete("/users/:id", (req, res) => {
+    const { id } = req.params
+
+    Users.remove(id)
+        .then(user => {
+            if (user) {
+                res.status(204).end()
+            } else {
+                res.status(404).json({ message: "User Not Found or Already Deleted" })
+            }
+        })
+        .catch(err => {
+            res.status(500).json({ message: "The user could not be removed" })
+        })
+})
+
+//PUT function to update a user
+server.put("/users/:id", (req, res) => {
+    const id = req.params.id;
+    const updatedUser = req.body;
+
+    if (!updatedUser.name || !updatedUser.bio) {
+        res.status(400).json({ errorMessage: "Please provide name and bio for the user." })
+    } else Users.update(id, updatedUser)
+        .then(updated => {
+            if (updated) {
+                res.status(200).json(updated);
```

Comment:
Here, on success, we are supposed to return the newly updated user document. However, User.update returns a 0 or 1 depending on whether or not a document was updated. So how are we to find the newly updated object? We'll use the id we got from the params instead:
```suggestion
                 Users.findById(id).then(user => res.status(200).json(user));
```